test/integration/apiserver/apply/apply_test.go

```diff
@@ -302,3 +309,70 @@ func TestApplyManagedFields(t *testing.T) {
 		t.Fatalf("Expected:\n%v\nGot:\n%v", string(expected), string(actual))
 	}
 }
+
+// TestApplyRemovesEmptyManagedFields there are no empty managers in managedFields
+func TestApplyRemovesEmptyManagedFields(t *testing.T) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	obj := []byte(`{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "test-cm",
+			"namespace": "default"
+		}
+	}`)
+
+	_, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		Namespace("default").
+		Resource("configmaps").
+		Name("test-cm").
+		Body(obj).
+		Do().
+		Get()
+	if err != nil {
+		t.Fatalf("Failed to create object using Apply patch: %v", err)
+	}
+
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		Namespace("default").
+		Resource("configmaps").
+		Name("test-cm").
+		Body(obj).Do().Get()
+	if err != nil {
+		t.Fatalf("Failed to patch object: %v", err)
+	}
+
+	object, err := client.CoreV1().RESTClient().Get().Namespace("default").Resource("configmaps").Name("test-cm").Do().Get()
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	accessor, err := meta.Accessor(object)
+	if err != nil {
+		t.Fatalf("Failed to get meta accessor: %v", err)
+	}
+
+	actual, err := json.MarshalIndent(object, "\t", "\t")
+	if err != nil {
+		t.Fatalf("Failed to marshal object: %v", err)
+	}
+
+	expected := []byte(`{
```

Comment:
I don't think we want another change detector. Can you check the presence (or lack of presence) of the managedFields in the object rather than compare the entire object?

Reply:
Sure